### PR TITLE
Add page not found pattern

### PR DESCRIPTION
**Pull request by alex-ju**
This PR is adding a new pattern for page not found pages.

The guidance and examples for this pattern were updated after [Page not found pages](https://github.com/alphagov/govuk-design-system-backlog/issues/130) has gone through the review session.

Authors: @stevenaproctor, @amyhupe 

[Preview changes](https://deploy-preview-291--govuk-design-system-preview.netlify.com)

### CODE REVIEW

File: src/patterns/page-not-found-pages/user-mistake/index.njk [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+layout: layout-example.njk
+---
+
+<div class="govuk-o-width-container">
+
+  <div class="govuk-o-main-wrapper">
```
Comment (igloosi) [inline src/patterns/page-not-found-pages/user-mistake/index.njk]: old class names here

File: src/patterns/page-not-found-pages/reason-not-known/index.njk [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            Name (optional)
+          </label>
+          <input class="govuk-c-input" id="name" name="name" type="text">
+          </div>
+          <div class="govuk-o-form-group"><label class="govuk-c-label" for="email">
+          Email address
+          </label>
+          <input class="govuk-c-input" id="email" name="email" type="text">
+          </div>
+          <button type="submit" class="govuk-c-button" id="continue">Report a page not found</button>
```
Comment (igloosi) [inline src/patterns/page-not-found-pages/reason-not-known/index.njk]: old class names here

File: src/patterns/page-not-found-pages/user-mistake/index.njk (hunk 1 above)
Comment (alex-ju) [reply src/patterns/page-not-found-pages/user-mistake/index.njk]: Updated the layout and components classes with the new ones.

File: src/patterns/page-not-found-pages/default/index.njk [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  <div class="govuk-o-main-wrapper">
+    <div class="govuk-o-grid">
+      <div class="govuk-o-grid__item govuk-o-grid__item--two-thirds">
+      
+        <h1 class="govuk-heading-xl">Page not found</h1>
+        <p class="govuk-body">We have reported this to the team that manages the service and they will fix it as soon as possible.</p>
+        <p class="govuk-body">Contact us if you need to speak to someone about your tax credits.</p>
+        <p class="govuk-body govuk-!-mb-r0">Telephone:</p>
+          <p class="govuk-body govuk-!-w-bold">0345 300 3900</p>
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply src/patterns/page-not-found-pages/default/index.njk]: Updated, indentation fixed

File: src/patterns/page-not-found-pages/reason-not-known/index.njk [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        <p class="govuk-body">If you typed the web address, check it is correct.</p>
+        <p class="govuk-body">If you pasted the web address, check you copied the entire address.</p>
+        <p class="govuk-body">If the web address is correct or you selected a link or button, you can:</p>
+          <ul class="govuk-list govuk-list--bullet">
+            <li><a class="govuk-list" href="https://www.gov.uk/contact-the-tax-credit-office">contact the Tax Credit Helpline</a> if you need to speak to someone about your tax credits</li>
+            <li>report the page not found</li>
+          </ul>
+        <p class="govuk-body">If you want us to tell you when we have fixed the link or button, give us your name and email address.</p>
+        <form action="/url/of/next/page" method="post">
+          <div class="govuk-o-form-group"><label class="govuk-c-label" for="name">
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply src/patterns/page-not-found-pages/reason-not-known/index.njk]: Updated to use macros where possbile (inputs and button)

**Comment by alex-ju** [comment]
@stevenaproctor: sure, you can update this branch with your changes or you can create a new branch from this and raise a separate pull request, however is easier for you.

**Comment by stevenaproctor** [comment]
@alex-ju I do not have push access to the repository so it might be easier for me to fork and do a separate PR once this one has been merged.

File: src/patterns/page-not-found-pages/default/index.njk [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  <div class="govuk-main-wrapper">
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds">
+
+        <h1 class="govuk-heading-xl">Page not found</h1>
+        <p class="govuk-body">We have reported this to the team that manages the service and they will fix it as soon as possible.</p>
+        <p class="govuk-body">Contact us if you need to speak to someone about your tax credits.</p>
+        <p class="govuk-body govuk-!-mb-r0">Telephone:</p>
+        <p class="govuk-body govuk-!-w-bold">0345 300 3900</p>
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply src/patterns/page-not-found-pages/default/index.njk]: Updated to use freephone numbers and removed the links to tax-credit website.

**Pull request merged**